crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -254,6 +254,19 @@ def prepare(options = {})
 
       protected
 
+      def finalize_cluster_upgrade(cluster)
+        save_upgrade_state("Finalizing the upgrade of nodes in cluster #{cluster}")
+
+        NodeObject.find("pacemaker_config_environment:#{cluster}").each do |node|
+          if node["pacemaker"]["founder"] && node.roles.include?("neutron-network")
+            node.run_ssh_cmd("source /root/.openrc; neutron-ha-tool --l3-agent-rebalance")
```

Comment:
There is an issue with relying on ```.openrc``` to be present  when calling any openstack cli during the upgrade. IIRC .openrc is only present on the node(s) that hold the keystone-server role. So in a multi-cluster setup the file might not be there. (We don't need to fix this now, but we should you track it somewhere)